### ISSUE TITLE
Hinted Compilation for modules

### ISSUE THREAD

**Issue by Make42** [Nuitka/NUITKA-Utilities#31]
I tried to use use hinted compiliation to compile my module with

```
python get-hints.py gp.py
python nuitka-hints.py --module gp.py
```

and got the error "conflicting options, cannot make standalone module".

Is this an issue regarding nuitka or nuitka-hinting? I would have thought that the hinting is just as usefull and applicable for modules as for executables.

**Comment by JorjMcKie** [comment]
Yes, this works as designed and documented.
If you want a module (i.e. Python extension), you cannot use hinted compilation. Otherwise leave out the ``--module`` parameter.

**Comment by JorjMcKie** [comment]
> Is this an issue regarding nuitka or nuitka-hinting? I would have thought that the hinting is just as usefull and applicable for modules as for executables.

No, it is not. If you use ``--module``, you will create a shared library (.PYD file extension on Windows), but otherwise it works like the Python source script. You may opt to include more Python sources in such a compile (via ``--recurse-to=``) and thus make the binary bigger, but if you don't do that, just your script gets compiled and needs all of your Python installation as before during execution.
